metis-indexing/metis-indexing-client/src/main/java/eu/europeana/indexing/client/config/IndexingConfig.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2007-2013 The Europeana Foundation
+ *
+ *  Licenced under the EUPL, Version 1.1 (the "Licence") and subsequent versions as approved
+ *  by the European Commission;
+ *  You may not use this work except in compliance with the Licence.
+ *
+ *  You may obtain a copy of the Licence at:
+ *  http://joinup.ec.europa.eu/software/page/eupl
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under
+ *  the Licence is distributed on an "AS IS" basis, without warranties or conditions of
+ *  any kind, either express or implied.
+ *  See the Licence for the specific language governing permissions and limitations under
+ *  the Licence.
+ */
+package eu.europeana.indexing.client.config;
+
+import com.mongodb.MongoClient;
+import com.mongodb.ServerAddress;
+import eu.europeana.indexing.service.PublishingService;
+import eu.europeana.indexing.service.dao.FullBeanDao;
+import eu.europeana.indexing.service.mongo.MorphiaDatastoreProvider;
+import eu.europeana.metis.exception.IndexingException;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import org.apache.commons.lang.StringUtils;
+import org.apache.solr.client.solrj.impl.CloudSolrServer;
+import org.apache.solr.client.solrj.impl.LBHttpSolrServer;
+
+public class IndexingConfig {
+	private static IndexingConfig instance = null;
+	private PublishingService publishingService;
+	private FullBeanDao fullBeanDao;
+	
+	protected IndexingConfig() throws IndexingException {
+		init();		
+	}
+	
+	public static IndexingConfig getInstance() throws IndexingException {
+		if (instance == null)
+			instance = new IndexingConfig();
+		
+		return instance;
+	}
+	
+	public PublishingService getPublishingService() {
+		return publishingService;
+	}
+	
+	public FullBeanDao getFullBeanDao() {
+		return fullBeanDao;
+	}
+	
+	private void init() throws IndexingException {
+		String resourceName = "indexing.properties";
+		ClassLoader loader = Thread.currentThread().getContextClassLoader();
+		Properties props = new Properties();
+
+		try(InputStream resourceStream = loader.getResourceAsStream(resourceName)) {
+		    props.load(resourceStream);
+		} catch (IOException e) {
+			throw new IndexingException("Could not load indexing configuration.", e.getCause());
+		}
+
+		// Mongo
+		String mongoHost = props.getProperty("mongo.host");
+		String mongoPort = props.getProperty("mongo.port");
+		String mongoUsername = props.getProperty("mongo.usernaname");

```

Comment:
This variable is never used! And neither are some other ones in the lines below!

Reply:
Need to check with Simon which fields exactly are required in production and how exactly the init logic should work, e.g. when to use Cloud Solr and when not, etc.